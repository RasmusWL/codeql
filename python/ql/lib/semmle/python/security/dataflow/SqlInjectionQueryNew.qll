private import codeql.queries.SharedSqlInjectionQuery
private import semmle.python.dataflow.new.internal.DataFlowImplSpecific::PythonDataFlow as PythonDataFlow
private import semmle.python.dataflow.new.internal.TaintTrackingImplSpecific::PythonTaintTracking as PythonTaintTracking
private import semmle.python.Concepts::PythonLangSig as PythonLangSigg

module Input implements InputSig<PythonDataFlow> {
  import semmle.python.Concepts
  import semmle.python.dataflow.new.RemoteFlowSources
  import semmle.python.dataflow.new.BarrierGuards
}

module Config =
  SqlInjectionQuery<PythonLangSigg, PythonDataFlow, PythonTaintTracking, Input>::SqlInjectionConfig;

module Flow =
  SqlInjectionQuery<PythonLangSigg, PythonDataFlow, PythonTaintTracking, Input>::SqlInjectionFlow;

module Query = SqlInjectionQuery<PythonLangSigg, PythonDataFlow, PythonTaintTracking, Input>;
