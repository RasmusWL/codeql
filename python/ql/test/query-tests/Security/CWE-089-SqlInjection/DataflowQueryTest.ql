import experimental.dataflow.TestUtil.DataflowQueryTest
import semmle.python.security.dataflow.SqlInjectionQueryNew::Config as Config
import FromTaintTrackingConfig<Config>
