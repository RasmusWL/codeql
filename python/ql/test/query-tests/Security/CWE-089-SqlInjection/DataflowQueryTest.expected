missingAnnotationOnSink
| sql_injection.py:17:24:17:64 | sql_injection.py:17 | ERROR, you should add `# $ MISSING: result=BAD` or `result=OK` annotation | "SELECT * FROM users WHERE username = %s" |
| sql_injection.py:18:26:18:66 | sql_injection.py:18 | ERROR, you should add `# $ MISSING: result=BAD` or `result=OK` annotation | "SELECT * FROM users WHERE username = %s" |
| sql_injection.py:40:26:40:68 | sql_injection.py:40 | ERROR, you should add `# $ MISSING: result=BAD` or `result=OK` annotation | "SELECT * FROM users WHERE username = '%s'" |
| sqlalchemy_textclause.py:28:31:28:34 | sqlalchemy_textclause.py:28 | ERROR, you should add `# $ MISSING: result=BAD` or `result=OK` annotation | stmt |
| sqlalchemy_textclause.py:33:31:33:34 | sqlalchemy_textclause.py:33 | ERROR, you should add `# $ MISSING: result=BAD` or `result=OK` annotation | stmt |
| sqlalchemy_textclause.py:37:31:37:34 | sqlalchemy_textclause.py:37 | ERROR, you should add `# $ MISSING: result=BAD` or `result=OK` annotation | stmt |
testFailures
| sql_injection.py:21:24:21:77 | ControlFlowNode for BinaryExpr | Unexpected result: result=BAD |
| sql_injection.py:24:38:24:95 | ControlFlowNode for BinaryExpr | Unexpected result: result=BAD |
| sql_injection.py:25:26:25:83 | ControlFlowNode for BinaryExpr | Unexpected result: result=BAD |
| sql_injection.py:26:28:26:85 | ControlFlowNode for BinaryExpr | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:27:28:27:87 | ControlFlowNode for Attribute() | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:31:50:31:72 | ControlFlowNode for Attribute() | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:41:26:41:33 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:42:31:42:38 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:43:30:43:37 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:44:35:44:42 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:45:41:45:48 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:46:46:46:53 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:47:47:47:54 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:48:52:48:59 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:50:18:50:25 | ControlFlowNode for username | Unexpected result: result=BAD |
| sqlalchemy_textclause.py:51:24:51:31 | ControlFlowNode for username | Unexpected result: result=BAD |
failures
